### PR TITLE
Fix #35: Add simpler example of state manipulation

### LINKED ISSUE #35: Request for a simple "counter" example in Tutorial
Working through the tutorial, I felt myself really wanting to see the most basic example of some UI changing its state. Perhaps a "counter" with a button, which when pressed, updates the count of some text.

IMHO, the Shopping Cart is a bit too complex as an example to show why we need StatefulComponents.

I'd be happy to help write this section if someone wants to contribute the canonical, simple "counter".

Thanks!

### PR DESCRIPTION
**Pull request by krisgiesing**
Also reworded surrounding text for clarity.

### CODE REVIEW

**Comment by krisgiesing** [comment]
@sethladd

**Comment by krisgiesing** [comment]
I sketched out some explanatory text but feel free to reword and polish.

File: tutorial.md [hunk 1]
```diff
@@ -218,20 +218,126 @@ message to the console. You can use `GestureDetector` to detect a variety of
...
-Change notifications, in contrast, flow "up" the component hierarchy by way of
-callbacks. For example, consider the `ShoppingListItem` widget below:
+In order to build more complex experiences, however - for example, to react
+in more interesting ways to user input - applications will typically carry some
+state. Flutter uses StatefulComponents to capture this idea. StatefulComponents
+are special widgets that know how to generate State objects, which are then
+used to hold state. Consider this basic example, using the `RaisedButton`
+mentioned earlier:
+
+```
```
Comment (abarth) [inline tutorial.md]: If you add the `dart` tag here, we'll get syntax highlighting (eventually)

**Comment by abarth** [comment]
Looks great.  LGTM

Would you be willing to fix up line 163 to use the parameter name "icon" rather than type?  That should make Travis green again.

**Pull request merged**